services/web/server/src/simcore_service_webserver/projects/_db_utils.py

```diff
@@ -335,28 +336,55 @@ async def _get_project(
         # this retrieves the projects where user is owner
         user_groups: list[RowProxy] = await self._list_user_groups(connection, user_id)
 
-        # NOTE: ChatGPT helped in producing this entry
-        conditions = sa.and_(
-            projects.c.uuid == f"{project_uuid}",
-            projects.c.type == f"{ProjectType.TEMPLATE.value}"
-            if only_templates
-            else True,
-            sa.or_(
-                projects.c.prj_owner == user_id,
-                sa.text(
-                    f"jsonb_exists_any(projects.access_rights, {assemble_array_groups(user_groups)})"
-                ),
-            ),
+        # Helper subquery that prepares access rights data in
+        # backwards compatible json type.
+        access_rights_subquery = (
+            select(
+                project_to_groups.c.project_uuid,
+                sa.func.jsonb_object_agg(
+                    project_to_groups.c.gid,
+                    sa.func.jsonb_build_object(
+                        "read",
+                        project_to_groups.c.read,
+                        "write",
+                        project_to_groups.c.write,
+                        "delete",
+                        project_to_groups.c.delete,
+                    ),
+                ).label("access_rights"),
+            )
+            .where(project_to_groups.c.project_uuid == f"{project_uuid}")
+            .group_by(project_to_groups.c.project_uuid)
+        ).subquery("access_rights_subquery")
+
+        query = (
+            sa.select(
+                *[col for col in projects.columns if col.name != "access_rights"],
+                access_rights_subquery.c.access_rights,
+            )
+            .select_from(projects.join(access_rights_subquery, isouter=True))
+            .where(
+                (projects.c.uuid == f"{project_uuid}")
+                & (
+                    projects.c.type == f"{ProjectType.TEMPLATE.value}"
+                    if only_templates
+                    else True
+                )
+            )
         )
 
         if only_published:
-            conditions &= projects.c.published == "true"
+            query = query.where(projects.c.published == "true")
 
-        query = select(projects).where(conditions)
         if for_update:
-            query = query.with_for_update()
-
-        result = await connection.execute(query)
+            # NOTE: It seems that blocking this row in the database is necessary; otherwise, there are some concurrency issues.
+            # As the WITH FOR UPDATE clause cannot be used with the GROUP BY clause, I have added a separate query for that.
+            blocking_query = (
+                sa.select(projects).where(projects.c.uuid == f"{project_uuid}")
+            ).with_for_update()
+            await connection.execute(blocking_query)
+
+        result = await connection.execute(query)  # <-- Here the query is executed
```

Comment:
MINOR: I guess the comment is not necessary ;-)

Reply:
You would be surprised! it is so hidden in the code that it saved me always at least 10 seconds while finding it :D

Reply:
but I will remove it